shell/utils/cluster.js

```diff
@@ -335,3 +335,132 @@ export async function initSchedulingCustomization(value, features, store, mode)
     clusterAgentDefaultPC, clusterAgentDefaultPDB, schedulingCustomizationFeatureEnabled, schedulingCustomizationOriginallyEnabled, errors
   };
 }
+
+/**
+ * Recursively filters a `diffs` object to only include differences that are relevant to the user.
+ * A difference is considered relevant if the user has provided a custom value for that specific field.
+ *
+ * @param {object} diffs - The object representing the differences between two chart versions' default values.
+ * @param {object} userVals - The object containing the user's custom values.
+ * @returns {object} A new object containing only the relevant differences.
+ */
+function _filterRelevantDifferences(diffs, userVals) {
```

Comment:
given these three functions aren't nicely grouped in a file / function can the name's be prefixed to make it clear they're all 1-1 connected?

_filterRelevantDifferences
_processAddonVersionChange
preserveAddonConfigs

becomes

_addonConfigPreserveFilter
_addonConfigPreserveProcess
addonConfigPreserve